File: solutions/go/isogram/2/isogram.go
// Package isogram should have a package comment that summarizes what it's about.
package isogram

import (
	"unicode"
)

// IsIsogram should have a comment documenting it.
func IsIsogram(word string) bool {
	letters := make(map[rune]bool)
	for _, letter := range word {
		if unicode.IsLetter(letter) {
			lower := unicode.ToLower(letter)
			if letters[lower] {
				return false
			}
			letters[lower] = true
		}
	}
	return true
}

File: solutions/zig/isogram/2/isogram.zig
const std = @import("std");

pub fn isIsogram(str: []const u8) bool {
    var letters: [26]bool = undefined;
    for (str) |c| {
        const i = switch (c) {
            'A'...'Z' => c - 'A',
            'a'...'z' => c - 'a',
            else => continue,
        };
        if (letters[i])
            return false;
        letters[i] = true;
    }
    return true;
}
